### ISSUE TITLE
Does img-size change the network architecture?

### ISSUE THREAD

**Issue by a-esp-1** [ultralytics/yolov5#1205]
## ❔Question
Hi,

I've trainning with 416x416 images. This images are crops from 10000x10000, 10017x10013 and 14000x14000 images. So I've tested my model with some of these 416x416 images.

My surprise comes when I run detect.py --img-size 10000 with one of this big images and the results are perfect.

So my question is **Does the --img-size value change the inputnetwork and all the arquitecture of the net? Does --img-size value rescale image dimensions because while running detect.py I can see WARNING: --img-size 10000 must be multiple of max stride 32, updating to 10016??**  

I mean results are perfect but I would like to understand what's going on because I'm using a 416x416 model with 10000x10000 images so something is happening here. One option is that img-size change the network architecture other option is that detect.py is cropping these big images in 416x416 images. I think this last option is not possible just by seeing detect.py code.

Thank you in advance.

**Comment by glenn-jocher** [comment]
@a-esp-1 YOLOv5 model architecture is fully defined by a model.yaml file such as https://github.com/ultralytics/yolov5/blob/master/models/yolov5s.yaml, and is not a function of input width and height.

https://github.com/ultralytics/yolov5/blob/7fe4a6bc06f0148e12edf71e296efc804778241a/models/yolov5s.yaml#L1-L48

**Comment by a-esp-1** [comment]
Thank you @glenn-jocher 

So if YOLOv5 architecture does not change, how can this be explained? I mean I've trainned a model with 416x416 images and I'm getting excelents results with 10000x10000 images when I run detect.py.

**Comment by a-esp-1** [comment]
I find the anwser here: https://stackoverflow.com/questions/62719641/why-pytorch-model-takes-multiple-image-size-inside-the-model

Maybe I explained my idea bad. What I mean is that the inputsize (--img-size value) of the net is addapted depending on the --img-size value. As I can see in https://github.com/ultralytics/yolov5/blob/master/detect.py line 62 (img = torch.zeros((1, 3, imgsz, imgsz), device=device)  # init img), the tensor depends on image size (--img-size value). As it's explained in the anwser I found (Stackoverflow) Pytorch adapts the net to this value. Architecture does not change depending on the img-size value, but it's adapted.
